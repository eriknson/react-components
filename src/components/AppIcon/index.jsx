import React from 'react'
import styled from 'styled-components'
import { devices } from '../theme'

const Icon = styled.img`
  height: auto;

  @media (max-width: ${devices.phone}) {
    width: 25px;
  }

  @media (min-width: ${devices.phone}) and (max-width: ${devices.tablet}) {
    width: 30px;
  }

  @media (min-width: ${devices.tablet}) {
    width: 40px;
  }
`

export default function AppIcon({ iconPath }) {
  return <Icon src={iconPath} />
}
