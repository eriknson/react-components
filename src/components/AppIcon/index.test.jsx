import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import ThemeProvider from '../ThemeProvider'
import theme from '../theme'
import AppIcon from './index'

describe('AppIcon', () => {
  afterEach(cleanup)
  test('renders the AppIcon', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <AppIcon />
      </ThemeProvider>
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
