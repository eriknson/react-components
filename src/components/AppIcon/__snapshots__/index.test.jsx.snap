// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppIcon renders the AppIcon 1`] = `
.c0 {
  height: auto;
}

@media (max-width:480px) {
  .c0 {
    width: 25px;
  }
}

@media (min-width:480px) and (max-width:868px) {
  .c0 {
    width: 30px;
  }
}

@media (min-width:868px) {
  .c0 {
    width: 40px;
  }
}

<img
  class="c0"
/>
`;
